pkg/web/apps/apps_psql_repository_test.go

```diff
@@ -1,56 +1,234 @@
+
+	if err != nil {
+		t.Fatalf("Got error trying to get apps from database: %v", err)
+	}
+
+	if len(*apps) != 2 {
+		t.Fatalf("Expected 2 apps to be returned from the database, got %v", len(*apps))
+	}
+}
 
+func Test_appsPostgreSQLRepository_GetApps_WillReturnNoApps(t *testing.T) {
+	db, mock, err := sqlmock.New()
 	if err != nil {
-		t.Errorf("Connect() returned an error: %v", err.Error())
-	}
-
-	tests := []struct {
-		name    string
-		want    *[]models.App
-		wantErr bool
-	}{
-		{
-			name: "appsPostgreSQLRepository.GetApps() should return a list of apps",
-			want: &[]models.App{
-				models.App{
-					ID:                    "a0874c82-2b7f-11e9-b210-d663bd873d93",
-					AppID:                 "com.aerogear.app1",
-					AppName:               "app1",
-					NumOfDeployedVersions: 1,
-					NumOfAppLaunches:      1,
-					NumOfClients:          1,
-				},
-			},
-			wantErr: false,
-		},
-	}
-
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			a := NewPostgreSQLRepository(dbConn)
-			got, err := a.GetApps()
-			if (err != nil) != tt.wantErr {
-				t.Errorf("appsPostgreSQLRepository.GetApps() error = %v, wantErr %v", err, tt.wantErr)
-			}
-			if !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("appsPostgreSQLRepository.GetApps() = %v, want %v", got, tt.want)
-			}
-		})
+		t.Fatalf("Unexpected error opening a stub database connection: %v", err)
+	}
+
+	defer db.Close()
+
+	mockApps := []models.App{
+		models.App{
+			ID:      "7f89ce49-a736-459e-9110-e52d049fc025",
+			AppID:   "com.aerogear.mobile_app_one",
+			AppName: "Mobile App One",
+		},
+		models.App{
+			ID:      "7f89ce49-a736-459e-9110-e52d049fc026",
+			AppID:   "com.aerogear.mobile_app_three",
+			AppName: "Mobile App Two",
+		},
+		models.App{
+			ID:      "7f89ce49-a736-459e-9110-e52d049fc027",
+			AppID:   "com.aerogear.mobile_app_three",
+			AppName: "Mobile App Three",
+		},
+	}
+
+	timestamp := "2019-02-15T09:38:33+00:00"
+
+	// Insert 3 apps which are soft deleted
+	sqlmock.NewRows([]string{"id", "app_id", "app_name", "deleted_at"}).AddRow(mockApps[0].ID, mockApps[0].AppID, mockApps[0].AppName, timestamp).AddRow(mockApps[1].ID, mockApps[1].AppID, mockApps[1].AppName, timestamp).AddRow(mockApps[2].ID, mockApps[2].AppID, mockApps[2].AppName, timestamp)
+
+	// We should expected 0 apps
+	mock.ExpectQuery(getAppsQueryString).WillReturnRows(&sqlmock.Rows{})
+	a := NewPostgreSQLRepository(db)
+
+	apps, err := a.GetApps()
+
+	if err != nil && err != models.ErrNotFound {
+		t.Fatalf("Expected ErrNotFound error to be returned from database, got %v", err)
+	}
+
+	if apps != nil && len(*apps) != 0 {
+		t.Fatalf("Expected 0 apps to be returned from the database, got %v", len(*apps))
+	}
+}
+
+func Test_appsPostgreSQLRepository_GetAppVersionsByAppID(t *testing.T) {
+	db, mock, err := sqlmock.New()
+	if err != nil {
+		t.Fatalf("Unexpected error opening a stub database connection: %v", err)
+	}
+
+	defer db.Close()
+
+	mockVersions := []models.Version{
```

Comment:
I know that you folks decided it and probably you will not change. However, I would like to say that I do not agree besides know that you will move forward in this way. The versions should not be the object of the app. They are different things. 
IHMO:
* The first page will use the endpoint /apps which should return a list of the object app. 
* then, is if someone clicks on in the app of the list the versions of that app with the totals should appear. So, it can't be the same object used in the list, it should be another object, for example, AppDetails with the attributes that are used in the details of an app.

Reply:
I do not understand your point @camilamacedo86 . The landing UI screen is a high-level list of all apps(`/apps`). The detailed UI screen is in the context of a single app (`/apps/{id}`) and all information including sub-information should be displayed in this screen.

Reply:
@camilamacedo86 nobody will see your message except me, you should CC the relevant people here or say this in Google Hangouts Chat so people are aware of your concerns.